### PR TITLE
feat #CCOP-110: possibility to save review from UI

### PR DESCRIPTION
**Pull request by YanBrovkin**

### CODE REVIEW

File: VirtoCommerce.Storefront/Helpers/SystemTime.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+using System;
+
+namespace VirtoCommerce.Storefront.Helpers
+{
+    public static class SystemTime
```
Comment (eMazeika) [inline VirtoCommerce.Storefront/Helpers/SystemTime.cs:5]: strange class. The time was never set?

File: VirtoCommerce.Storefront/Controllers/Api/ApiCatalogController.cs [hunk 2]
```diff
@@ -67,5 +75,13 @@ public async Task<ActionResult<Category[]>> GetCategoriesByIds(string[] category
         {
             return await _catalogService.GetCategoriesAsync(categoryIds, respGroup);
         }
+
+        // POST: storefrontapi/product/{id}/review
```
Comment (eMazeika) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCatalogController.cs]: incorrect comment

File: VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs [hunk 3]
```diff
@@ -360,6 +360,8 @@ protected virtual Task LoadProductCustomerReviewsAsync(List<Product> products, W
                     };
                     return _customerReviewService.SearchReviews(criteria);
                 }, 1, CustomerReviewSearchCriteria.DefaultPageSize);
+
+                product.AverageRating = Convert.ToInt32(product.CustomerReviews?.Average(r => r.Rating) ?? 0);
```
Comment (eMazeika) [inline VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs]: it's incorrect to calculate it in storefront, not in module

**Review by eMazeika — COMMENTED** [review]
Could product variations share the reviews?

File: VirtoCommerce.Storefront/Helpers/SystemTime.cs (hunk 1 above)
Comment (YanBrovkin) [reply VirtoCommerce.Storefront/Helpers/SystemTime.cs:5]: Helper class created especially for tests, e.g. for verification results with DateTime.Now

File: VirtoCommerce.Storefront/Controllers/Api/ApiCatalogController.cs (hunk 2 above)
Comment (YanBrovkin) [reply VirtoCommerce.Storefront/Controllers/Api/ApiCatalogController.cs]: fixed

File: VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs (hunk 3 above)
Comment (YanBrovkin) [reply VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs]: In which module do you propose to calculate it ? In CustomerReviews ?

Comment (YanBrovkin) [reply VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs]: fixed

**Pull request merged**